From 86dd7a672a078e56a0498e408fd9a70c40176dd7 Mon Sep 17 00:00:00 2001
From: "Lopez Esparza, Ricardo" <ricardo.lopez.esparza@intel.com>
Date: Tue, 19 Apr 2016 19:18:46 -0500
Subject: [PATCH 08/41] ov8858: improve W/A to sync exposure settings.

The previous fix uses a sleep function not accurate for
the time needed, a better solution is provided.

Change-Id: I9d7d31a3096c88258c6383ae1a6f50cc8326903b
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-31785
Signed-off-by: Lopez Esparza, Ricardo <ricardo.lopez.esparza@intel.com>
Reviewed-on: https://android.intel.com:443/498781
Reviewed-on: https://android.intel.com:443/504751
---
 drivers/external_drivers/camera/drivers/media/i2c/ov8858.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/drivers/external_drivers/camera/drivers/media/i2c/ov8858.c b/drivers/external_drivers/camera/drivers/media/i2c/ov8858.c
index 306d6f6..81c9c2d 100644
--- a/drivers/external_drivers/camera/drivers/media/i2c/ov8858.c
+++ b/drivers/external_drivers/camera/drivers/media/i2c/ov8858.c
@@ -419,6 +419,11 @@ static int ov8858_set_exposure(struct v4l2_subdev *sd, int exposure, int gain,
 	const struct ov8858_resolution *res;
 	u16 hts, vts;
 	int ret;
+	/* W/A: In CHT_MRD there is a sync problem between the new exposure,
+	 * and the statistics being reported to AE. This delay allows the old
+	 * statistics to be correctly reported before applying a new exposure */
+	if (strcmp(dmi_get_system_info(DMI_BOARD_NAME), CHT_HR_DEV_NAME) != 0)
+		usleep_range(4000, 6000);
 
 	mutex_lock(&dev->input_lock);
 
-- 
1.9.1

